controllers/front/paymentreturn.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+* Team Paylike
+*
+*  @author    Team Paylike
+*  @copyright Team Paylike
+*  @license   MIT license: https://opensource.org/licenses/MIT
+*/
+
+class PaylikePaymentReturnModuleFrontController extends ModuleFrontController
+{
+	public function __construct()
+	{
+		parent::__construct();
+		$this->context = Context::getContext();
+	}
+
+	public function init()
+	{
+		parent::init();
+		$cart = $this->context->cart;
+		$paylike = new Paylike();
+		$total = $cart->getOrderTotal(true, Cart::BOTH);
+		$customer = new Customer((int)$cart->id_customer);
+		$currency = new Currency((int)$cart->id_currency);
+		$paylikeapi = new PaylikeAPI(Configuration::get('PAYLIKE_SECRET_KEY'));
+		$amount = Tools::ps_round($total, 2) * 100;
+
+		if (Configuration::get('PAYLIKE_CHECKOUT_MODE') == 'delayed')
+			$capture = $paylikeapi->transactions->fetch(Tools::getValue('transactionid'));
+		else
+		{
+			$capture = $paylikeapi->transactions->capture(Tools::getValue('transactionid'),
+			[
+			'currency' => $currency->iso_code,
+			'amount' => $amount,
+			]);
+		}
+
+		if ($capture->transaction->currency == $currency->iso_code && $capture->transaction->custom->cartId == $cart->id && $capture->transaction->amount == $amount)
```

Comment:
Authenticating and checking currency (Order will be created in Prestashop after authenticating)